lib/upnp/igd_service.go

```diff
@@ -97,10 +203,10 @@ func (s *IGDService) DeletePortMapping(ctx context.Context, protocol nat.Protoco
 // GetExternalIPAddress queries the IGD service for its external IP address.
 // Returns nil if the external IP address is invalid or undefined, along with
 // any relevant errors
-func (s *IGDService) GetExternalIPAddress(ctx context.Context) (net.IP, error) {
-	tpl := `<u:GetExternalIPAddress xmlns:u="%s" />`
+func (s *IGDService) GetExternalIPv4Address(ctx context.Context) (net.IP, error) {
+	const template = `<u:GetExternalIPAddress xmlns:u="%s" />`
 
-	body := fmt.Sprintf(tpl, s.URN)
+	body := fmt.Sprintf(template, s.URN)
```

Comment:
Same simplifications regarding unmarshaling variables etc. could be applied here in `GetExternalIPv4Address` and in `AddPortMapping` above.